cli/src/client.ts

```diff
@@ -150,21 +163,45 @@ export class Client {
 
   async getPrice(): Promise<Price> {
     try {
-      const { data } = await axios.get<Pool>(
+      const { data: ethPool } = await axios.get<Pool>(
+        'https://daemon.thorchain.shapeshift.com/lcd/thorchain/pool/ETH.ETH',
+      )
+
+      const { data: foxPool } = await axios.get<Pool>(
         'https://daemon.thorchain.shapeshift.com/lcd/thorchain/pool/ETH.FOX-0XC770EEFAD204B5180DF6A14EE197D99D808EE52D',
       )
 
-      const runeInAsset = new BigNumber(data.balance_asset).div(data.balance_rune)
-      const assetPriceUsd = new BigNumber(data.asset_tor_price)
-        .div(new BigNumber(10).pow(THORCHAIN_PRECISION))
+      const ethPriceUsd = toPrecision(ethPool.asset_tor_price, THORCHAIN_PRECISION).toFixed(THORCHAIN_PRECISION)
+      const foxPriceUsd = toPrecision(foxPool.asset_tor_price, THORCHAIN_PRECISION).toFixed(THORCHAIN_PRECISION)
+      const runeInAsset = new BigNumber(foxPool.balance_asset).div(foxPool.balance_rune)
+      const runePriceUsd = runeInAsset.times(foxPriceUsd).toFixed(THORCHAIN_PRECISION)
+
+      const address = UNIV2_ETH_FOX_LP_TOKEN
+      const abi = parseAbi([
+        'function getReserves() view returns (uint112 _reserve0, uint112 _reserve1, uint32 _blockTimestampLast)',
+        'function totalSupply() view returns (uint256 totalSupply)',
+      ])
+
+      const reserves = await this.rpc.readContract({ address, abi, functionName: 'getReserves' })
```

Comment:
worth adding a comment that token0 is always WETH and token1 is always FOX due to sorting?

Reply:
sure, I will link to the contract for ref: 3b920bc14417209f94644745b113391deef9b2f8

I think the return type is mostly implied by viem usage, so going to leave it as such (re: sorting)